src/blockdata/witness.rs

```diff
@@ -0,0 +1,209 @@
+use consensus::{Decodable, Encodable, WriteExt, serialize};
+use VarInt;
+use io::{self, Read, Write};
+use consensus::encode::Error;
+use prelude::*;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]  //TODO manually implement?
+pub struct Witness {
+
+    /// contains the witness Vec<Vec<u8>> serialization without the initial varint indicating the
+    /// number of elements (which is stored in len)
+    content: Vec<u8>,
```

Comment:
If we're going down this path, does it make sense to also consider something like Bitcoin Core's "prevec" for the witness contents?

Reply:
FWIW: https://docs.rs/tinyvec/1.5.0/tinyvec/enum.TinyVec.html

Reply:
Yea, I mean such a thing is super-duper trivial to write.

Reply:
I think a tinyvec like structure could probably be applicable broadly (like also for inputs/outpus) and as an addition to this, probably better to address in another PR